flink-runtime/src/main/java/org/apache/flink/runtime/io/network/metrics/CreditBasedInputBufferPoolMetric.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.io.network.metrics;
+
+import org.apache.flink.metrics.Gauge;
+import org.apache.flink.runtime.io.network.partition.consumer.InputChannel;
+import org.apache.flink.runtime.io.network.partition.consumer.RemoteInputChannel;
+import org.apache.flink.runtime.io.network.partition.consumer.SingleInputGate;
+
+/**
+ * Gauge metric measuring the input buffer pool usage gauge for {@link SingleInputGate}s under credit based mode.
+ */
+public class CreditBasedInputBufferPoolMetric {
+
+	/**
+	 * Gauge metric measuring the floating buffer usage for {@link SingleInputGate}.
+	 */
+	public static class FloatingBufferPoolUsage implements Gauge<Float> {
+
+		public SingleInputGate[] inputGates;
+
+		public FloatingBufferPoolUsage(SingleInputGate[] inputGates) {
+			this.inputGates = inputGates;
+		}
+
+		@Override
+		public Float getValue() {
+			int leftFloatingBuffer = 0;
+			int requestedFloatingBuffer = 0;
+
+			for (SingleInputGate inputGate : inputGates) {
+				requestedFloatingBuffer += inputGate.getBufferPool().bestEffortGetNumOfUsedBuffers();
+				for (InputChannel ic : inputGate.getInputChannels().values()) {
+					if (ic instanceof RemoteInputChannel) {
+						leftFloatingBuffer += ((RemoteInputChannel) ic).unsafeGetFloatingBufferLeft();
+					}
+				}
+			}
+
+			if (requestedFloatingBuffer == 0) {
+				return 0.0f;
+			} else {
+				return 1 - ((float) leftFloatingBuffer) / requestedFloatingBuffer;
```

Comment:
I think it is not very accurate here. 
It should be `(requestedFloatingBuffer - leftFloatingBuffer) / LocalBufferPool.getNumBuffers`